### DISCUSSION TITLE
Q&A 37.1 — ParticleSwarmOptimizer: Configuration, Convergence & Edge Cases

### DISCUSSION THREAD

**Discussion by web3guru888** [web3guru888/asi-build#761] [Q&A]
### Configuration Questions

**Q1: How should inertia weight decay be configured?**
Linear decay from ω_max=0.9 to ω_min=0.4 over max_iterations is standard (Shi & Eberhart 1998). The formula:
```
ω(t) = ω_max − (ω_max − ω_min) · t / max_iterations
```
Alternatively, use constriction factor χ (Clerc & Kennedy 2002) which provides theoretical convergence guarantees without manual ω tuning.

**Q2: When should I use constriction factor vs. inertia weight?**
- **Inertia weight**: More intuitive tuning, good for problems where you want explicit exploration→exploitation control
- **Constriction factor**: Mathematically grounded convergence guarantee, fewer hyperparameters, recommended as default

**Q3: How is velocity clamping Vmax set?**
Rule of thumb: `Vmax = k · (upper_bound - lower_bound)` where k ∈ [0.1, 0.5]. Too low restricts exploration; too high causes oscillation.

**Q4: How does MOPSO handle multiple objectives?**
External archive stores non-dominated (Pareto-optimal) solutions. Leader selection uses adaptive grid (hypercubes) in objective space — leaders chosen from less crowded regions to maintain diversity along the Pareto front.

### Edge Cases

**Q5: What if all particles converge to the same position?**
Population diversity collapse — diversity metric drops below threshold. Mitigations:
1. Velocity clamping prevents zero-velocity stagnation
2. Adaptive topology can switch from gbest→lbest to slow information flow
3. Turbulence operator adds random perturbation to worst particles

**Q6: How to handle constraint violations in bounded search?**
Position clamping: `x_i = clip(x_i, lower, upper)`. Velocity reflection: if position hits bound, reverse velocity component. Absorbing walls vs. reflecting walls trade-off.

Ref: Issue #755 | Wiki: Phase-37-Particle-Swarm-Optimizer
